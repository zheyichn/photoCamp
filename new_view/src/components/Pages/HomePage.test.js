import { render, screen, cleanup, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import Homepage from "./HomePage";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import Home from "@mui/icons-material/Home";

describe("Homepage Component should render correctly", () => {
  test("main view inside homepage", () => {
    render(<Homepage></Homepage>);
    const mainDiv = screen.getByTestId("main-view");
    expect(mainDiv).toBeInTheDocument();
  });

  it("Homepage matches snapshot", () => {
    const tree = renderer.create(<Homepage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
