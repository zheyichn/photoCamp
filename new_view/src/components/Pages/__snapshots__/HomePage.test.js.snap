// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Component should render correctly Homepage matches snapshot 1`] = `
<div>
  <div>
    <div
      className="MuiGrid-root css-54nrqh-MuiGrid-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
          style={
            {
              "backgroundColor": "white",
            }
          }
        >
          <div
            className="MuiGrid-root MuiGrid-item css-fom513-MuiGrid-root"
            style={
              {
                "backgroundColor": "white",
              }
            }
          >
            <div
              className="MuiTypography-root MuiTypography-h6 css-fd7qc2-MuiTypography-root"
            >
              PhotoCamp
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container css-raykiu-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-x1qdbo-MuiGrid-root"
    >
      <main
        data-testid="main-view"
      >
        <span
          className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
          role="progressbar"
          style={
            {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={{}}
            />
          </svg>
        </span>
      </main>
    </div>
  </div>
</div>
`;
